00_Base/src/services/credentials.service.ts

```diff
@@ -474,6 +478,109 @@ export class CredentialsService {
     return updatedClientInformation;
   }
 
+  async generateCredentialsTokenA(
+    credentialsRequest: CredentialsRequestDTO,
+    versionNumber: VersionNumber,
+  ): Promise<CredentialsDTO> {
+    const serverCredentialsRoleDTOs = credentialsRequest.roles;
+
+    // Make sure we stored the necessary version and version endpoints
+    // so that MSP can retrieve them later in the registration process
+    const storedVersion = await this.versionRepository.readOnlyOneByQuery(
```

Comment:
can we move this to a reusable validator, or better yet maybe we can do this in middleware? It would be nice to have a single method that can validate that the version number is supported for a given ModuleId...

Reply:
Hi @elliot-sabitov, thanks for the feedback! I updated the code in this commit: 897073ec23ae680316e4f6b7ca76fdc50eee4544. Please feel free to let me know if there is any further comments.

Regarding this version validation, I move it partially into a validator script since I still need all version endpoints in the following step. We might not need to do it in a middleware for every module additionally. Because version module has no delete actions defined in OCPI protocol. As long as cpo and msp exchange version endpoints that each other has during the registration, it should be safe.